src/sentry/workflow_engine/processors/detector.py

```diff
@@ -268,18 +270,24 @@ def evaluate_group_key_value(
             metrics.incr("workflow_engine.detector.skipping_invalid_condition_group")
             return None
 
+        # TODO: We need to handle tracking consecutive evaluations before emitting a result here. We're able to
+        # store these in `DetectorStateData.counter_updates`, but we don't have anywhere to set the required
+        # thresholds at the moment. Probably should be a field on the Detector? Could also be on the condition
+        # level, but usually we want to set this at a higher level.
         new_status = DetectorPriorityLevel.OK
+        is_group_condition_met, condition_results = evaluate_condition_group(
+            self.condition_group, value, conditions=self.conditions
+        )
 
-        for condition in self.conditions:
-            # TODO: We need to handle tracking consecutive evaluations before emitting a result here. We're able to
-            # store these in `DetectorStateData.counter_updates`, but we don't have anywhere to set the required
-            # thresholds at the moment. Probably should be a field on the Detector? Could also be on the condition
-            # level, but usually we want to set this at a higher level.
-            evaluation = condition.evaluate_value(value)
+        if is_group_condition_met:
+            validated_condition_results: list[DetectorPriorityLevel] = [
+                DetectorPriorityLevel(int(result))
```

Comment:
is the `condition_result` for a `DataCondition` on a `StatefulDetector` ever not an `int`?

Reply:
i think it should actually always be a detector priority level tbh.

i think this can actually just be: `[result for result in condition_results if result is not None]` (i'll update)

Reply:
the only reason it's casting everything like this is to appease mypy 😭 -- instead of doing this much work though, i'll just check to see if it's `not None` and `isinstance(result, DetectorPriorityLevel)` (that's quite a bit less runtime logic to appease the type checks)